packages/services/src/transactions/stacks-transactions.utils.ts

```diff
@@ -0,0 +1,17 @@
+import { StacksTx } from '@leather.io/models';
+
+export function removeConfirmedTransactions(
+  mempoolTransactions: StacksTx[],
+  confirmedTransactions: StacksTx[]
+): StacksTx[] {
+  const confirmedTxIds = new Set(confirmedTransactions.map(tx => tx.tx_id));
+  return mempoolTransactions.filter(tx => !confirmedTxIds.has(tx.tx_id));
+}
+
+export function removeStaleTransactions(
+  mempoolTransactions: StacksTx[],
+  confirmedTransactions: StacksTx[]
+): StacksTx[] {
+  const confirmedNonces = new Set(confirmedTransactions.map(tx => tx.nonce));
+  return mempoolTransactions.filter(tx => !confirmedNonces.has(tx.nonce));
+}
```

Comment:
When touching filtering logic in extension recently, I refactored some filtering code that was similar to this.

This API decision of accepting array directly leads to deep nesting of calls, which gets kinda ugly esp. when line wrapped e.g. 

```ts
const txs = removeStaleTransactions(
  mempoolTxs,
  removeConfirmedTransactions(mempoolTxs, confirmedTxs)
);
```

Returning higher order filter functions helps remedy this, and make it more composable


```ts
const txs = mempoolTransactions
  .filter(filterOutConfirmedTxs(confirmedTransactions))
  .filter(filterOutStaleTransactions(confirmedTransactions))
```